Sources/Sentry/SentryProfiler.mm

```diff
@@ -319,6 +325,20 @@ + (BOOL)isRunning
 #    endif // SENTRY_TARGET_PROFILING_SUPPORTED
 }
 
+#    pragma mark - Testing
+
++ (void)useSystemWrapper:(SentrySystemWrapper *)systemWrapper
+{
+    std::lock_guard<std::mutex> l(_gProfilerLock);
+    _gCurrentSystemWrapper = systemWrapper;
+}
+
++ (void)useProcessInfoWrapper:(SentryNSProcessInfoWrapper *)processInfoWrapper
+{
+    std::lock_guard<std::mutex> l(_gProfilerLock);
+    _gCurrentProcessInfoWrapper = processInfoWrapper;
+}
+
```

Comment:
`l`: Instead of adding two extra methods, we could also pass these in as parameters for the initializer. Then we don't need the test category, and we don't need to init these properties below on lines 486 to 497.

Reply:
That was actually my first approach, but because of the current interface of SentryProfiler being only class methods with no external init, it was going to be a bigger refactor to do so.